ql/experimental/barrieroption/mcdoublebarrierengine.hpp

```diff
@@ -0,0 +1,311 @@
+                                                         maxSamples_);
+            results_.value = this->mcModel_->sampleAccumulator().mean();
+            if (RNG::allowsErrorEstimate)
+            results_.errorEstimate =
+                this->mcModel_->sampleAccumulator().errorEstimate();
+        }
+      protected:
+        // McSimulation implementation
+        TimeGrid timeGrid() const;
+        ext::shared_ptr<path_generator_type> pathGenerator() const {
+            TimeGrid grid = timeGrid();
+            typename RNG::rsg_type gen =
+                RNG::make_sequence_generator(grid.size()-1,seed_);
+            return ext::shared_ptr<path_generator_type>(
+                         new path_generator_type(process_,
+                                                 grid, gen, brownianBridge_));
+        }
+        ext::shared_ptr<path_pricer_type> pathPricer() const;
+        // data members
+        ext::shared_ptr<GeneralizedBlackScholesProcess> process_;
+        Size timeSteps_, timeStepsPerYear_;
+        Size requiredSamples_, maxSamples_;
+        Real requiredTolerance_;
+        bool antithetic_;
+        bool brownianBridge_;
+        BigNatural seed_;
+    };
+
+    //! Monte Carlo double-barrier-option engine factory
+    template <class RNG = PseudoRandom, class S = Statistics>
+    class MakeMCDoubleBarrierEngine {
+      public:
+        explicit MakeMCDoubleBarrierEngine(
+                    const ext::shared_ptr<GeneralizedBlackScholesProcess>&);
+        // named parameters
+        MakeMCDoubleBarrierEngine& withSteps(Size steps);
+        MakeMCDoubleBarrierEngine& withStepsPerYear(Size steps);
+        MakeMCDoubleBarrierEngine& withBrownianBridge(bool b = true);
+        MakeMCDoubleBarrierEngine& withAntitheticVariate(bool b = true);
+        MakeMCDoubleBarrierEngine& withSamples(Size samples);
+        MakeMCDoubleBarrierEngine& withAbsoluteTolerance(Real tolerance);
+        MakeMCDoubleBarrierEngine& withMaxSamples(Size samples);
+        MakeMCDoubleBarrierEngine& withSeed(BigNatural seed);
+        // conversion to pricing engine
+        operator ext::shared_ptr<PricingEngine>() const;
+      private:
+        ext::shared_ptr<GeneralizedBlackScholesProcess> process_;
+        bool brownianBridge_, antithetic_;
+        Size steps_, stepsPerYear_, samples_, maxSamples_;
+        Real tolerance_;
+        BigNatural seed_;
+    };
+
+    class DoubleBarrierPathPricer : public PathPricer<Path> {
+      public:
+        DoubleBarrierPathPricer(
+                    DoubleBarrier::Type barrierType,
+                    Real barrierLow,
+                    Real barrieHigh,
+                    Real rebate,
+                    Option::Type type,
+                    Real strike,
+                    const std::vector<DiscountFactor>& discounts,
+                    const ext::shared_ptr<StochasticProcess1D>& diffProcess,
+                    const PseudoRandom::ursg_type& sequenceGen);
```

Comment:
Not sure what these two parameters are for?  `diffProcess_` is not used, and `sequenceGen_` is used to create a random vector `u` which in turn is unused.

Reply:
Have removed those variables.